docs/python.md

```diff
@@ -627,6 +649,230 @@ explorer.find_function("foo").find_return().find_comp_ifs()[0].is_equivalent("i
+len(try_stmt.find_excepts()) # 3
+```
+
+`is_equivalent()` cannot be used on the items found by `find_excepts()` because a standalone `except` raises a `SyntaxError`.
+
+#### `find_except`
+
+Returns a specific except handler matching the exception type and optional variable name.
+
+```python
+code_str = """
+try:
+    x = 1 / 0
+except ZeroDivisionError:
+    print("division by zero")
+except ValueError as e:
+    print(f"value error: {e}")
+except:
+    print("other error")
+"""
+node = Node(code_str)
+try_stmt = node.find_trys()[0]
+try_stmt.find_except("ZeroDivisionError").find_body().is_equivalent("print('division by zero')") # True
+try_stmt.find_except("ValueError", "e").find_body().is_equivalent("print(f'value error: {e}')") # True
+try_stmt.find_except().find_body().is_equivalent("print('other error')") # True (bare except)
+```
+
+#### `has_except`
+
+Returns True if a try statement has an except handler for the given exception type and variable name.
+
+```python
+code_str = """
+try:
+    x = 1 / 0
+except ZeroDivisionError:
+    print("division by zero")
+except ValueError as e:
+    print(f"value error: {e}")
+"""
+node = Node(code_str)
+try_stmt = node.find_trys()[0]
+try_stmt.has_except("ZeroDivisionError") # True
+try_stmt.has_except("ValueError", "e") # True
+try_stmt.has_except("ValueError") # False
+```
+
+#### `find_try_else`
+
+Returns the else block of a try statement.
+
+```python
+code_str = """
+try:
+    x = 1
+except ValueError:
+    print("error")
+else:
+    print("success")
+    x = 2
+"""
+node = Node(code_str)
+try_stmt = node.find_trys()[0]
+try_stmt.find_try_else().is_equivalent("print('success')\nx = 2") # True
+```
+
+#### `find_finally`
+
+Returns the finally block of a try statement.
+
+```python
+code_str = """
+try:
+    x = 1
+except ValueError:
+    print("error")
+finally:
+    print("cleanup")
+    x = None
+"""
+node = Node(code_str)
+try_stmt = node.find_trys()[0]
+try_stmt.find_finally().is_equivalent("print('cleanup')\nx = None") # True
+```
+
+#### `find_matches`
+
+Returns a list of match statements.
+
+```python
+code_str = """
+match x:
+  case 0:
+    pass
+  case _:
+    pass
+
+match y:
+  case 1:
+    pass
+"""
+node = Node(code_str)
+len(node.find_matches()) # 2
+node.find_matches()[0].is_equivalent("match x:\n  case 0:\n    pass\n  case _:\n    pass") # True
+```
+
+#### `find_match_subject`
+
+Returns the subject of a match statement.
+
+```python
+code_str = """
+match x:
+  case 0:
+    pass
+"""
+node = Node(code_str)
+node.find_matches()[0].find_match_subject().is_equivalent("x") # True
+```
+
+#### `find_match_cases`
+
+Returns a list of case blocks in a match statement.
+
+```python
+code_str = """
+match x:
+  case 0:
+    print(0)
+    print('spam')
+  case _:
+    pass
+"""
+node = Node(code_str)
+len(node.find_matches()[0].find_match_cases()) # 2
+node.find_matches()[0].find_match_cases()[0].find_case_body().is_equivalent("print(0)\nprint('spam')") # True
```

Comment:
Got this error when running this snippet:

```bash
AttributeError: 'Node' object has no attribute 'find_case_body'. Did you mean: 'find_case_guard'?
```

Reply:
I removed `find_case_body`. It should be replaced by the generic `find_body`